asyncpg/connection.py

```diff
@@ -1041,6 +1032,23 @@ def terminate(self):
         if reset_query:
             await self.execute(reset_query, timeout=timeout)
 
+    def _abort(self):
+        # Put the connection into the aborted state.
+        self._aborted = True
+        self._protocol.abort()
```

Comment:
I'd also add `self._protocol = None` to catch any use of connection after `_abort()`.